Runtime/Scripts/Texture2DDisposable.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace Unity.MeshSync {
+    /// <summary>
+    /// Wrapper class for Texture2D to handle cleaning up resources when the texture is
+    /// no longer required.
+    /// </summary>
+    internal class Texture2DDisposable : IDisposable {
+        public Texture2D Texture { get; private set; }
+
+        private bool DoDispose { get; }
+
+        public Texture2DDisposable(Texture2D texture, bool doDispose = true) {
+            Init(texture);
+            this.DoDispose = doDispose;
+        }
+
+        private void Init(Texture2D texture) {
+            Texture = texture;
+        }
+
+        public void ReadFromRenderTexture(RenderTexture renderTexture) {
+            var activeRenderTarget = RenderTexture.active;
+
+            RenderTexture.active = renderTexture;
+            Texture.ReadPixels(new Rect(0, 0, renderTexture.width, renderTexture.height), 0, 0);
+            Texture.Apply();
+
+            RenderTexture.active = activeRenderTarget;
+        }
```

Comment:
Is this code used anywhere? since the Texture getter is public, you could have an extension on Texture2D to copy from RenderTexture

Reply:
This is used by `SaveMaterialRenderTexturesToAssetDatabase`. I don't think it's good to make it an extension method because then the user needs to reference the Texture of the object instead of keeping the logic inside this class.